src/test/util/transaction_utils.h

```diff
@@ -6,6 +6,10 @@
 #define BITCOIN_TEST_UTIL_TRANSACTION_UTILS_H
 
 #include <primitives/transaction.h>
+#include <coins.h>
+#include <script/signingprovider.h>
```

Comment:
I don't think you need the includes in the header. Forward decls should be enough if passing in references.

Reply:
@theStack I am happy to merge either version. Just let me know if you plan to address this comment or not.

Reply:
@MarcoFalke: I'm working on it right now, should be ready in a bit.